plugins/defaultplugins/ifplugin/interface_config.go

```diff
@@ -673,37 +667,35 @@ Modify rx-mode on specified VPP interface
 */
 func (plugin *InterfaceConfigurator) modifyRxModeForInterfaces(oldIntf *intf.Interfaces_Interface, newIntf *intf.Interfaces_Interface,
 	ifIdx uint32) error {
-	oldRxSettings := oldIntf.RxModeSettings
-	newRxSettings := newIntf.RxModeSettings
-	if oldRxSettings != newRxSettings {
-		var oldRxMode intf.RxModeType
-		if oldRxSettings != nil {
-			oldRxMode = oldRxSettings.RxMode
-		}
-		if newRxSettings != nil {
+	oldRx := oldIntf.RxModeSettings
+	newRx := newIntf.RxModeSettings
+
+	if oldRx == nil && newRx != nil || oldRx != nil && newRx == nil ||
+		oldRx.RxMode != newRx.RxMode || oldRx.QueueId != newRx.QueueId || oldRx.QueueIdValid != newRx.QueueIdValid {
```

Comment:
You could shorten the part of this comparison which compares fields to: `if ... || *oldRx != *newRx {`. This will compare all the fields of the struct.